management/backup/locations_test.go

```diff
@@ -203,6 +203,166 @@ func TestListLocations(t *testing.T) {
+				}
+
+				if req.PMMServerConfig != nil {
+					require.NotNil(t, loc.PMMServerConfig)
+					assert.Equal(t, req.PMMServerConfig.Path, loc.PMMServerConfig.Path)
+				}
+
+				if req.PMMClientConfig != nil {
+					require.NotNil(t, loc.PMMClientConfig)
+					assert.Equal(t, req.PMMClientConfig.Path, loc.PMMClientConfig.Path)
+				}
+
+				if req.S3Config != nil {
+					require.NotNil(t, loc.S3Config)
+					assert.Equal(t, req.S3Config.Endpoint, loc.S3Config.Endpoint)
+					assert.Equal(t, req.S3Config.AccessKey, loc.S3Config.AccessKey)
+					assert.Equal(t, req.S3Config.SecretKey, loc.S3Config.SecretKey)
+				}
+
+				found = true
+
+				break
+			}
+		}
+		assert.True(t, found)
+	}
+	t.Run("update name and config path", func(t *testing.T) {
+		t.Parallel()
+
+		addReqBody := locations.AddLocationBody{
+			Name:        gofakeit.Name(),
+			Description: gofakeit.Question(),
+			PMMServerConfig: &locations.AddLocationParamsBodyPMMServerConfig{
+				Path: "/tmp",
+			},
+		}
+		resp, err := client.AddLocation(&locations.AddLocationParams{
+			Body:    addReqBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		defer deleteLocation(t, client, resp.Payload.LocationID)
+
+		updateBody := locations.ChangeLocationBody{
+			LocationID: resp.Payload.LocationID,
+			Name:       gofakeit.Name(),
+			PMMServerConfig: &locations.ChangeLocationParamsBodyPMMServerConfig{
+				Path: "/tmp/nested",
+			},
+		}
+		_, err = client.ChangeLocation(&locations.ChangeLocationParams{
+			Body:    updateBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+
+		listResp, err := client.ListLocations(&locations.ListLocationsParams{Context: pmmapitests.Context})
+		require.NoError(t, err)
+
+		checkChange(t, updateBody, listResp.Payload.Locations)
+	})
+
+	t.Run("change config type", func(t *testing.T) {
+		t.Parallel()
+
+		addReqBody := locations.AddLocationBody{
+			Name:        gofakeit.Name(),
+			Description: gofakeit.Question(),
+			PMMServerConfig: &locations.AddLocationParamsBodyPMMServerConfig{
+				Path: "/tmp",
+			},
+		}
+		resp, err := client.AddLocation(&locations.AddLocationParams{
+			Body:    addReqBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		defer deleteLocation(t, client, resp.Payload.LocationID)
+
+		updateBody := locations.ChangeLocationBody{
+			LocationID: resp.Payload.LocationID,
+			Name:       gofakeit.Name(),
+			S3Config: &locations.ChangeLocationParamsBodyS3Config{
+				Endpoint:  "https://example.com",
+				AccessKey: "access_key",
+				SecretKey: "secret_key",
+			},
+		}
+		_, err = client.ChangeLocation(&locations.ChangeLocationParams{
+			Body:    updateBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+
+		listResp, err := client.ListLocations(&locations.ListLocationsParams{Context: pmmapitests.Context})
+		require.NoError(t, err)
+
+		checkChange(t, updateBody, listResp.Payload.Locations)
```

Comment:
I think you should check that `PMMServerConfig` is set to nil after `ChangeLocation` call.

Reply:
Maybe even better to check that all configs are nil except `S3Config`.